prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirToHl7Converter.kt

```diff
@@ -0,0 +1,192 @@
+package gov.cdc.prime.router.fhirengine.translation.hl7
+
+import ca.uhn.hl7v2.HL7Exception
+import ca.uhn.hl7v2.model.Message
+import ca.uhn.hl7v2.util.Terser
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.ConfigSchema
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.ConfigSchemaElement
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.ConfigSchemaReader
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.FhirPathUtils
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.HL7Utils
+import org.apache.logging.log4j.kotlin.Logging
+import org.hl7.fhir.r4.model.Base
+import org.hl7.fhir.r4.model.Bundle
+
+/**
+ * Convert a FHIR [bundle] to an HL7 message using the [schemaRef] to perform the conversion.
+ * The converter will error out if [strict] is set to true and there is an error during the conversion.  if [strict]
+ * is set to false (the default) then any conversion errors are logged as a warning.  Note [strict] does not affect
+ * the schema validation process.
+ * @property terser the terser to use for building the HL7 message (use for dependency injection)
+ */
+class FhirToHl7Converter(
+    private val bundle: Bundle,
+    private val schemaRef: ConfigSchema,
+    private val strict: Boolean = false,
+    private var terser: Terser? = null
+) : Logging {
+    /**
+     * Convert a FHIR [bundle] to an HL7 message using the [schema] in the [schemaFolder] location to perform the conversion.
+     * The converter will error out if [strict] is set to true and there is an error during the conversion.  if [strict]
+     * is set to false (the default) then any conversion errors are logged as a warning.  Note [strict] does not affect
+     * the schema validation process.
+     * @property terser the terser to use for building the HL7 message (use for dependency injection)
+     */
+    constructor(
+        bundle: Bundle,
+        schema: String,
+        schemaFolder: String,
+        strict: Boolean = false,
+        terser: Terser? = null
```

Comment:
Kotlin question - If adding the ? makes terser nullable, what is the reason for setting the = null as the back up option if it is null?

Reply:
What Cle said.  If you look at the data classes for the config, it all has default values due to Jackson requiring a default constructor ().